<a href="https://colab.research.google.com/github/Balqis-Famuyon/Machine_Learning_-Model/blob/main/Balqis_Famuyon.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This project shows a a macjine learning model is being built for the prediction of a flower specie

Importing libraries


In [1]:
import pandas as pd
import numpy as np

Exploring the data


In [3]:
from sklearn.datasets import load_iris
iris_dataset = load_iris()

In [4]:
print('keys of iris_dataset:\n{}'.format(iris_dataset.keys()))

keys of iris_dataset:
dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])


Checking for Data Description


In [5]:
print(iris_dataset['DESCR'][:193] + '\n...')

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, pre
...


Exploring the Target Name Column

In [6]:
print('Target names: {}'.format(iris_dataset['target_names']))

Target names: ['setosa' 'versicolor' 'virginica']


Checking Data type

In [7]:
print('Type of data:{}'.format(type(iris_dataset['data'])))

Type of data:<class 'numpy.ndarray'>


Viewing the descriptions of Measurement

In [8]:
print('Feature names:{}'.format(iris_dataset['feature_names']))

Feature names:['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


Checking for Data type and size

In [9]:
print('Type of data:{}'.format(type(iris_dataset['data'])))
print('Shape of data:{}'.format(iris_dataset['data'].shape))

Type of data:<class 'numpy.ndarray'>
Shape of data:(150, 4)


In [10]:
print('First five columns:\n{}'.format(iris_dataset['data'][:5]))

First five columns:
[[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]


In [11]:
print('Type of target:{}'.format(type(iris_dataset['target'])))

Type of target:<class 'numpy.ndarray'>


In [13]:
print('Shape of target:{}'.format(iris_dataset['target'].shape))

Shape of target:(150,)


In [14]:
print('Target:\n{}:'.format(iris_dataset['target']))

Target:
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]:


The target numbers displayed above depicts the specie of the flower. 0 represents setosa , 1 represents ersicolor ,2 represents virginica.

##Training and Testing Data
This project aims to build a machine learning model that can predict the species of iris for a new set of measurements.
The training data is going to be different feom the testing data.

> Indented block



In [15]:
import sklearn
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(
    iris_dataset['data'], iris_dataset['target'], random_state=0)

The output of the code above will make 75% of the data set the training data and the remaining 25% as the testing data

In [17]:
print('X_train shape: {}'.format(X_train.shape))
print('y_train shape: {}'.format(y_train.shape))
print('X_test shape: {}' .format(X_test.shape))
print('y_test shape: {}'.format(y_test.shape))

X_train shape: (112, 4)
y_train shape: (112,)
X_test shape: (38, 4)
y_test shape: (38,)


Building the Machine Learning Model

In [29]:
from sklearn.neighbors import KNeighborsClassifier
Knn = KNeighborsClassifier(n_neighbors=1)

"Knn" is the name of the model that is bring built , it hols the algorithm of this project. the k nearest neighbor for this model is 1 (K=1).

In [30]:
Knn.fit(X_train , y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

Making Predictions
To Predict a specie of any iris plant that i come across, i would use the feature properties of the plant- sepal length(5cm), sepal width
(2.9cm) , petal length(1cm), petal width(0.2cm).
to display the feature data in a Numpyarray, the shape has to be calculated- number of samples multiplied by the number of features; the number of samples in this scenario is 1. 

In [31]:
X_new = np.array([[5 , 2.9 ,1 , 0.2]])
print('X_new.shape: {}'.format(X_new.shape))

X_new.shape: (1, 4)


In [37]:
prediction = Knn.predict(X_new)

print('Prediction: {}'. format(prediction))
print('Predicted target name: {}'.format(
iris_dataset['target_names'][prediction]))

Prediction: [0]
Predicted target name: ['setosa']


Evaluating the Model

In [39]:
y_pred = Knn.predict(X_test)
print('Test set Prediction :\n {}'.format(y_pred))

Test set Prediction :
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


In [40]:
print('Test set score: {}'.format(np.mean(y_pred==y_test)))

Test set score: 0.9736842105263158


The Test score shown above depicts a 97% accuracy of the machone learning model that was built.